### PR TITLE
gnu-smalltalk: optional emacs support

### PR DESCRIPTION
**Pull request by joachifm**
Works for me.

@svenkeidel

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -3574,7 +3574,7 @@ let
     inherit fetchurl stdenv;
   });
 
-  gnu-smalltalk = callPackage ../development/compilers/gnu-smalltalk { };
+  gnu-smalltalk = callPackage ../development/compilers/gnu-smalltalk (config.gnu-smalltalk or {});
```
Comment (edolstra) [inline pkgs/top-level/all-packages.nix]: Please use
```
... { emacsSupport = config.gnu-smalltalk.emacsSupport or false; }
```

Comment (joachifm) [inline pkgs/top-level/all-packages.nix]: Done, but may I ask why? I've seen it used elsewhere and it seems a bit cleaner to me than setting individually.

Comment (lethalman) [inline pkgs/top-level/all-packages.nix]: Why @edolstra ?

**Comment by lethalman** [comment]
Don't use `config` for package-specific options please. I could understand `config.emacsSupport`, but not `config.gnu-smalltalk`.

**Comment by joachifm** [comment]
Done

**Comment by lethalman** [comment]
Ok thanks.

**Pull request merged**